Tests/kaas/k8s-node-distribution/k8s-node-distribution-check.py

```diff
@@ -0,0 +1,252 @@
+    ]
+
+    async with kubernetes_asyncio.client.ApiClient() as api:
+        core_api = kubernetes_asyncio.client.CoreV1Api(api)
+        result = await core_api.list_node()
+        for item in result.items:
+            node = dict()
+            node_labels = item.metadata.labels
+            for label in labels:
+                try:
+                    node[label] = node_labels[label]
+                except KeyError:
+                    # Node is not labelled as a control plane node
+                    pass
+            nodes.append(node)
+
+    return nodes
+
+
+def compare_labels(node_list, node_type="master"):
+
+    labels = {
+        "topology.kubernetes.io/region": set(),
+        "topology.kubernetes.io/zone": set(),
+        "kubernetes.io/hostname": set(),
+    }
+
+    for node in node_list:
+        for key in labels:
+            labels[key].add(node[key])
+
+    for label in labels:
+        if len(labels[label]) <= 1:
+            if len(labels[label]) == 1:
+                logger.warning(f"There seems to be no distribution across multiple {label.split('/')[1]}s "
+                               "or labels aren't set correctly across nodes.")
+            elif len(labels[label]) == 0:
+                logger.warning(f"The label for {label.split('/')[1]}s don't seem to be set for all nodes.")
+        else:
+            logger.info(f"The nodes are distributed across {str(len(labels[label]))} {label.split('/')[1]}s.")
+            return
+
+    if node_type == "master":
+        raise DistributionException()
+    elif node_type == "worker":
+        logger.warning("No node distribution could be detected for the worker nodes. "
+                       "This produces only a warning, since it is just a recommendation.")
+        return
+
+
+async def main(argv):
+    try:
+        config = initialize_config(parse_arguments(argv))
+    except (OSError, ConfigException, HelpException) as e:
+        if hasattr(e, 'message'):
+            logger.error(e.message)
+        print_usage()
+        return 1
+
+    nodes = await get_k8s_cluster_labelled_nodes(config.kubeconfig)
+
+    if len(nodes) < 2:
+        logger.error("The tested cluster only contains a single node, which can't comply with the standard.")
+        return 2
+
+    labelled_master_nodes = [node for node in nodes if "node-role.kubernetes.io/control-plane" in node]
+    try:
+        if len(labelled_master_nodes) > 1 and len(nodes) > 2:
+            worker_nodes = [node for node in nodes if "node-role.kubernetes.io/control-plane" not in node]
+            # Compare the labels of both types, since we have enough of them with labels
+            compare_labels(labelled_master_nodes, "master")
+            compare_labels(worker_nodes, "worker")
+        else:
+            compare_labels(nodes)
+    except DistributionException:
+        logger.error("The distribution of nodes described in the standard couldn't be detected.")
```

Comment:
Preferably pass this string literal as argument to the exception upon throwing it; here just use `str(exc)` or something similar.

Reply:
Understandable, done in the fixup.